src/main.rs

```diff
@@ -36,41 +37,62 @@ async fn main() -> Result<(), ()> {
         .unwrap_or_exit()
         .0;
 
+    let data_dir =
+        create_data_dir().map_err(|e| println!("Error creating LNDK's data dir {e:?}"))?;
+    setup_logger(config.log_level, config.log_dir)?;
+
     let creds = validate_lnd_creds(
         config.cert_path,
         config.cert_pem,
         config.macaroon_path,
         config.macaroon_hex,
     )
     .map_err(|e| {
-        println!("Error validating config: {e}.");
+        error!("Error validating config: {e}.");
     })?;
     let address = config.address.clone();
     let lnd_args = LndCfg::new(config.address, creds.clone());
 
     let (shutdown, listener) = triggered::trigger();
-    let signals = LifecycleSignals { shutdown, listener };
+    let signals = LifecycleSignals {
+        shutdown: shutdown.clone(),
+        listener: listener.clone(),
+    };
     let args = Cfg {
         lnd: lnd_args,
         signals,
         skip_version_check: config.skip_version_check,
     };
 
+    let mut sigterm_stream = tokio::signal::unix::signal(SignalKind::terminate())
+        .map_err(|e| error!("Error initializing sigterm signal: {e}."))?;
+    let mut sigint_stream = tokio::signal::unix::signal(SignalKind::interrupt())
+        .map_err(|e| error!("Error initializing sigint signal: {e}."))?;
+
+    tokio::spawn(async move {
+        tokio::select! {
+            _ = sigint_stream.recv() => {
+                info!("Received CTRL-C, shutting down..");
+                shutdown.trigger();
+            }
+            _ = sigterm_stream.recv() => {
+                info!("Received SIGTERM, shutting down..");
+                shutdown.trigger();
+            }
+        }
+    });
```

Comment:
Maybe we can add an issue to get integration tests for this up? I can look into it then.

Reply:
I had actually already gotten pretty far into writing a basic integration test for this, before realizing it was probably too big for this PR heh. I'll either put it up as a draft or finish it off. But I have some further integration test ideas I'll post as issues today, if you happen to be interested.